Mathlib/Tactic/WorkOnGoal.lean

```diff
@@ -12,14 +12,14 @@ namespace Mathlib.Tactic
 open Lean Elab.Tactic
 
 /--
-`work_on_goal n tacSeq` creates a block scope for the `n`-th goal (indexed from zero) and
-tries the sequence of tactics `tacSeq` on it.
+`work_on_goal n tacSeq` creates a block scope for the `n`-th goal and tries the sequence
+of tactics `tacSeq` on it.
 
 The goal is not required to be solved and any resulting subgoals are inserted back into the
 list of goals, replacing the `n`-th goal.
 -/
 elab (name := workOnGoal) "work_on_goal " n:num ppSpace seq:tacticSeq : tactic => do
-  match (← getGoals).splitAt n.toNat with
+  match (← getGoals).splitAt (n.toNat - 1) with
     | (_, []) => throwError "not enough goals"
```

Comment:
This means work_on_goal 0 and work_on_goal 1 do the same thing.

Reply:
Yeah, for more context see [this](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60work_on_goal.60.20vs.20.60swap.60/near/270587312)

Reply:
Right, I'm aware. Maybe document that this is intended usage, at least in the docstring here.

Reply:
Will do when I get home

Reply:
I think you should throw an error (either a parse error or a tactic error) for `0` entry, to prevent people from getting confused. Also, this should either be backported, or mathport has to accomodate the indexing change. (Both are not too difficult)